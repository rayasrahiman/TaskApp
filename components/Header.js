import React from 'react';
import {View, Image, StyleSheet, Dimensions} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {Colors} from '../constants/colors';

export default function Header() {
  return (
    <View style={styles.container}>
      <Image
        source={require('../assets/Logo.png')}
        style={styles.img}
        resizeMode="contain"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomColor: Colors.gray700,
    borderBottomWidth: wp(0.3),
    width: Dimensions.get('window').width,
  },
  img: {
    width: wp(15),
    height: hp(15),
  },
});
