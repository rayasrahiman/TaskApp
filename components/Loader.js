import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React from 'react';
import {heightPercentageToDP as hp} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export default function Loader() {
  return (
    <View style={styles.container}>
      <ActivityIndicator color={Colors.primary700} size={hp(4)} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
