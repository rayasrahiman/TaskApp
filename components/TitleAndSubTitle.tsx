import {View, Text, StyleSheet, Platform} from 'react-native';
import React from 'react';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export type Props = {
  title: string;
  subTitle: string;
  titleContProp: any;
  titleTxtProp: any;
  subTitleContProp: any;
  emailSubTitleContProp: any;
  subTitleTxtProp: any;
  viewPager: boolean;
  email: boolean;
  emailVal: string;
};

const TitleAndSubTitle: React.FC<Props> = ({
  title,
  subTitle,
  titleContProp,
  titleTxtProp,
  subTitleContProp,
  emailSubTitleContProp,
  subTitleTxtProp,
  viewPager,
  email,
  emailVal,
  ...titleAndSubTitleProp
}) => {
  return (
    <View {...titleAndSubTitleProp}>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...titleContProp,
        }}>
        <Text style={{...styles.titleTxt, ...titleTxtProp}}>{title}</Text>
      </View>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...subTitleContProp,
        }}>
        <Text
          style={{
            ...styles.subTitleTxt,
            ...{color: Colors.grayishBlack},
            ...subTitleTxtProp,
          }}>
          {subTitle}
        </Text>
      </View>
      {email && (
        <View
          style={{
            ...{
              marginHorizontal: viewPager
                ? 0
                : Platform.OS === 'android'
                ? wp(5)
                : wp(3),
              marginBottom: hp(2),
            },
            ...emailSubTitleContProp,
          }}>
          <Text
            style={{
              ...styles.subTitleTxt,
              ...{color: Colors.black},
              ...subTitleTxtProp,
            }}>
            {emailVal}
          </Text>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  titleTxt: {
    color: Colors.darkBlue,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(5) : hp(4),
  },
  subTitleTxt: {
    color: Colors.grayishBlack,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
});

export default TitleAndSubTitle;
