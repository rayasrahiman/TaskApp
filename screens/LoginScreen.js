import React, {useState} from 'react';
import {
  View,
  Text,
  KeyboardAvoidingView,
  Platform,
  Dimensions,
  ScrollView,
  StyleSheet,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import Header from '../components/Header';
import Button from '../components/Button';
import Input from '../components/Input';
import TitleAndSubTitle from '../components/TitleAndSubTitle';
import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function LoginScreen({navigation, route}) {
  const [disable, setDisable] = useState(true);
  const [error, setError] = useState(false);
  const {email} = route.params;

  const regex =
    /^(\S)(?=.*[0-9])(?=.*[A-Z])(?=.*[a-z])(?=.*[~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_₹])[a-zA-Z0-9~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_₹]{8,20}$/;

  const validate = text => {
    setDisable(!regex.test(text));
  };

  const redirect = () => {
    if (!disable) {
      navigation.navigate('Dashboard');
    } else {
      setError(true);
    }
  };

  return (
    <View style={styles.container}>
      <Header />
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.container}>
        <ScrollView>
          <TitleAndSubTitle
            title="Looks like you already have a Pentair account!"
            subTitle="Sign in to go to your account."
            subTitleContProp={styles.subTitleCont}
          />
          <Input
            editable={false}
            label={I18n.t('Email')}
            placeholder={I18n.t('Email')}
            value={email}
            inputValue={true}
          />
          <Input
            password
            editable={true}
            label={I18n.t('Password')}
            placeholder={I18n.t('PasswordTxt')}
            onChangeText={validate}
            error={error && <Text>{I18n.t('PasswordError')}</Text>}
            onFocus={() => setError(false)}
          />

          <Button
            testID="signIn"
            btnContainerProp={{
              backgroundColor: disable ? Colors.primary400 : Colors.primary700,
            }}
            title={I18n.t('Signin')}
            btnWrapper={{...styles.btnCont, ...styles.btnFirst}}
            onPress={redirect}
          />
          <Button
            title={I18n.t('SwitchAccount')}
            btnWrapper={{...styles.btnCont, ...styles.btnSecond}}
            btnContainerProp={styles.btnSecondCont}
            btnTxtProp={styles.btnSecondTxt}
            onPress={() => {}}
          />
        </ScrollView>
      </KeyboardAvoidingView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  subTitleCont: {
    marginVertical: hp(2),
  },
  btnCont: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    justifyContent: Platform.OS === 'ios' ? 'flex-end' : null,
    alignItems: Platform.OS === 'ios' ? 'flex-end' : null,
  },
  btnFirst: {
    marginTop:
      Platform.OS === 'android' && Dimensions.get('window').height < 700
        ? hp(8.5)
        : hp(12.5),
  },
  btnSecond: {
    marginVertical: hp(3),
    marginHorizontal: wp(5),
  },
  btnSecondCont: {
    backgroundColor: 'white',
    borderColor: Colors.primary700,
    borderWidth: wp(0.3),
  },
  btnSecondTxt: {
    color: Colors.primary700,
  },
});
