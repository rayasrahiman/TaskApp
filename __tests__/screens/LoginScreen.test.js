import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {shallow} from 'enzyme';
import {useNavigation} from '@react-navigation/native';

import LoginScreen from '../../screens/LoginScreen';
import Header from '../../components/Header';
import TitleAndSubTitle from '../../components/TitleAndSubTitle';
import Loader from '../../components/Loader';

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(),
  };
});

describe('LoginScreen', () => {
  const mockedParams = {
    route: {params: {email: 'test@test.com'}},
  };
  test('should render Login screen correctly', () => {
    const wrapper = render(<LoginScreen {...mockedParams} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Header', () => {
    const wrapper = shallow(<LoginScreen {...mockedParams} />);
    wrapper.find(<Header />);
  });

  test('should render TitleAndSubTitle component', () => {
    const wrapper = shallow(<LoginScreen {...mockedParams} />);
    wrapper.find(<TitleAndSubTitle />);
  });

  test('should render Loader component', () => {
    const wrapper = shallow(<LoginScreen {...mockedParams} />);
    wrapper.find(<Loader />);
  });

  test('should render TitleAndSubTitle texts', () => {
    const wrapper = render(<LoginScreen {...mockedParams} />);
    wrapper.getByText('Looks like you already have a Pentair account!');
    wrapper.getByText('Sign in to go to your account.');
  });

  test('should render label', () => {
    const wrapper = render(<LoginScreen {...mockedParams} />);
    wrapper.getByText('Sign In');
    wrapper.getByText('Switch Account');
    wrapper.getByText('Forgot Password');
  });

  test('should show invalid message when invalid password entered', () => {
    const {getByTestId, getByText} = render(<LoginScreen {...mockedParams} />);
    fireEvent.press(getByTestId('signIn'));
    getByText(
      'Must contain at least one uppercase, lowercase, numeric character, special character and Length must be between 8 to 20 characters.',
    );
  });

  test('should navigate to Dashboard Screen.', async () => {
    const mockNavigate = jest.fn();
    useNavigation.mockReturnValueOnce({navigate: mockNavigate});

    const wrapper = render(<LoginScreen {...mockedParams} />);

    const button = wrapper.getByTestId('signIn');
    fireEvent.press(button);

    await (() => {
      expect(mockNavigate).toHaveBeenCalledWith('Dashboard');
    });
  });
});
