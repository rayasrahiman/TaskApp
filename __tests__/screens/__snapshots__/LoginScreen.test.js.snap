// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render Login screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#221c30",
        "borderBottomWidth": 2.5,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/Logo.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 112.5,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View>
          <View
            style={
              Object {
                "marginHorizontal": 22.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#0570c9",
                  "fontSize": 53.5,
                  "fontWeight": "bold",
                }
              }
            >
              Looks like you already have a Pentair account!
            </Text>
          </View>
          <View
            style={
              Object {
                "marginHorizontal": 22.5,
                "marginVertical": 26.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#777",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                }
              }
            >
              Sign in to go to your account.
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "marginHorizontal": 22.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                }
              }
            >
              Email Address
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#888",
                "borderColor": "#221c30",
                "borderRadius": 13.5,
                "borderWidth": 2.5,
                "flexDirection": "row",
                "height": 80,
                "marginHorizontal": 22.5,
                "marginVertical": 10,
                "paddingHorizontal": 15,
              }
            }
          >
            <TextInput
              autoCorrect={false}
              editable={false}
              onBlur={[Function]}
              onFocus={[Function]}
              placeholder="Email Address"
              style={
                Object {
                  "color": "#221c30",
                  "flex": 1,
                  "fontSize": 26.5,
                }
              }
              value="test@test.com"
            />
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "marginHorizontal": 22.5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                }
              }
            >
              Password
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#221c30",
                "borderRadius": 13.5,
                "borderWidth": 2.5,
                "flexDirection": "row",
                "height": 80,
                "marginHorizontal": 22.5,
                "marginVertical": 10,
                "paddingHorizontal": 15,
              }
            }
          >
            <TextInput
              autoCorrect={false}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Enter password"
              secureTextEntry={true}
              style={
                Object {
                  "color": "#0570c9",
                  "flex": 1,
                  "fontSize": 26.5,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                allowFontScaling={false}
                onPress={[Function]}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 40,
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ó°›‘
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginHorizontal": 22.5,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#0570c9",
                  "flex": 1,
                  "fontSize": 26.5,
                }
              }
            >
              Forgot Password
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "flex-end",
              "marginHorizontal": 37.5,
              "marginTop": 167,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#44bdf5",
                "borderRadius": 10,
                "height": 93.5,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="signIn"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                  "paddingVertical": 6.5,
                }
              }
            >
              Sign In
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "flex-end",
              "marginHorizontal": 37.5,
              "marginVertical": 40,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#0570c9",
                "borderRadius": 10,
                "borderWidth": 2.5,
                "height": 93.5,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#0570c9",
                  "fontSize": 40,
                  "paddingVertical": 6.5,
                }
              }
            >
              Switch Account
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
